// bold-ui/lib/components/Modal/Modal.stories.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var addon_actions_1 = require("@storybook/addon-actions");
var addon_knobs_1 = require("@storybook/addon-knobs");
var react_1 = require("@storybook/react");
var react_2 = __importDefault(require("react"));
var Button_1 = require("../Button");
var HFlow_1 = require("../HFlow");
var auto_1 = require("./auto");
var ModalMountTarget_1 = require("./auto/ModalMountTarget");
var Modal_1 = require("./Modal");
var ModalBody_1 = require("./ModalBody");
var ModalContainer_1 = require("./ModalContainer");
var ModalFooter_1 = require("./ModalFooter");
var sizes = {
    large: 'large',
    small: 'small',
    auto: 'auto',
};
react_1.storiesOf('Components|Modal', module)
    .add('default', function () { return (react_2.default.createElement(Modal_1.Modal, { open: addon_knobs_1.boolean('open', true), size: addon_knobs_1.select('size', sizes, 'large'), onClose: addon_actions_1.action('close') },
    react_2.default.createElement(ModalBody_1.ModalBody, null,
        react_2.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer molestie, dui id luctus commodo, nunc enim facilisis odio, vel hendrerit erat neque eu nisl. Donec urna felis, pharetra nec urna a, efficitur lobortis urna. Mauris varius purus vehicula lorem mollis, a cursus enim malesuada. Integer at congue enim. Nullam purus mauris, fermentum nec mattis in, cursus nec tellus. Nunc sodales orci tortor, at feugiat purus hendrerit a. Suspendisse potenti. Nam porta urna vitae nibh pharetra eleifend. Nullam urna eros, auctor vitae maximus non, feugiat eget odio. Cras venenatis, lectus eget consectetur volutpat, urna felis efficitur enim, vitae viverra purus risus sed purus."),
        react_2.default.createElement("p", null, "Aliquam placerat eget massa vel congue. Suspendisse libero erat, auctor ac tristique et, ultrices quis dolor. Suspendisse commodo lacinia lectus ut egestas. Cras in urna eget leo mollis luctus eu eget felis. Mauris vel libero vel lectus cursus placerat. Nunc non ligula urna. Aliquam id quam at quam sodales gravida condimentum quis eros. am dolor nisl, ullamcorper ac varius nec, vulputate sed metus. Cras lacus ante, dapibus facilisis risus at, varius commodo nulla. Nunc volutpat pellentesque mauris et sodales. Suspendisse finibus, dui at tristique mollis, felis mauris vulputate leo, sed vehicula lacus nibh ut dolor.")),
    react_2.default.createElement(ModalFooter_1.ModalFooter, null,
        react_2.default.createElement(HFlow_1.HFlow, { justifyContent: 'flex-end' },
            react_2.default.createElement(Button_1.Button, { onClick: addon_actions_1.action('cancel clicked') }, "Cancel"),
            react_2.default.createElement(Button_1.Button, { kind: 'primary', onClick: addon_actions_1.action('save clicked') }, "Save"))))); })
    .add('auto', function () { return (react_2.default.createElement(react_2.default.Fragment, null,
    react_2.default.createElement(Button_1.Button, { onClick: auto_1.modal({
            size: 'small',
            render: function () { return 'Confirm?'; },
            actions: [
                { label: 'Cancel', onClick: addon_actions_1.action('Cancel') },
                { label: 'Ok', kind: 'primary', onClick: addon_actions_1.action('Ok') },
            ],
        }) }, "Auto modal"),
    react_2.default.createElement(ModalMountTarget_1.ModalMountTarget, null))); })
    .add('parts', function () { return (react_2.default.createElement(ModalContainer_1.ModalContainer, { onClose: addon_actions_1.action('onClose') },
    react_2.default.createElement(ModalBody_1.ModalBody, null, "Teste"),
    react_2.default.createElement(ModalFooter_1.ModalFooter, null,
        react_2.default.createElement(HFlow_1.HFlow, { justifyContent: 'flex-end' },
            react_2.default.createElement(Button_1.Button, null, "Secondary"),
            react_2.default.createElement(Button_1.Button, { kind: 'primary' }, "Primary"))))); })
    .add('modal overlap', function () { return (react_2.default.createElement(react_2.default.Fragment, null,
    react_2.default.createElement(Modal_1.Modal, { open: true, size: addon_knobs_1.select('main modal size', sizes, 'large'), onClose: addon_actions_1.action('close') },
        react_2.default.createElement(ModalBody_1.ModalBody, null,
            react_2.default.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer molestie, dui id luctus commodo, nunc enim facilisis odio, vel hendrerit erat neque eu nisl. Donec urna felis, pharetra nec urna a, efficitur lobortis urna. Mauris varius purus vehicula lorem mollis, a cursus enim malesuada. Integer at congue enim. Nullam purus mauris, fermentum nec mattis in, cursus nec tellus. Nunc sodales orci tortor, at feugiat purus hendrerit a. Suspendisse potenti. Nam porta urna vitae nibh pharetra eleifend. Nullam urna eros, auctor vitae maximus non, feugiat eget odio. Cras venenatis, lectus eget consectetur volutpat, urna felis efficitur enim, vitae viverra purus risus sed purus.")),
        react_2.default.createElement(ModalFooter_1.ModalFooter, null,
            react_2.default.createElement(HFlow_1.HFlow, { justifyContent: 'flex-end' },
                react_2.default.createElement(Button_1.Button, { onClick: addon_actions_1.action('cancel clicked') }, "Cancel"),
                react_2.default.createElement(Button_1.Button, { kind: 'primary', onClick: auto_1.modal({
                        size: 'small',
                        render: function () { return 'Confirm?'; },
                        depthLevel: 2,
                        manageOverflow: false,
                        onClose: addon_actions_1.action('close'),
                        actions: [
                            { label: 'Cancel', onClick: addon_actions_1.action('Cancel') },
                            { label: 'Ok', kind: 'primary', onClick: addon_actions_1.action('Ok') },
                        ],
                    }) }, "Open a modal that overlaps")))),
    react_2.default.createElement(ModalMountTarget_1.ModalMountTarget, null))); });
