// bold-ui/lib/components/Grid/Grid.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var addon_knobs_1 = require("@storybook/addon-knobs");
var react_1 = require("@storybook/react");
var react_2 = __importDefault(require("react"));
var __1 = require("../");
var useBreakpoint_1 = require("../../hooks/useBreakpoint");
var useWindowSize_1 = require("../../hooks/useWindowSize");
var styles = {
    box: {
        backgroundColor: '#eee',
        textAlign: 'center',
        height: '100%',
    },
};
var justifyContentOptions = [
    'flex-start',
    'flex-end',
    'center',
    'space-between',
    'space-around',
    'space-evenly',
];
var alignItemsOptions = ['flex-start', 'flex-end', 'center', 'baseline', 'stretch'];
var alignSelfOptions = ['auto', 'flex-start', 'flex-end', 'center', 'baseline', 'stretch'];
var directionOptions = ['row', 'row-reverse', 'column', 'column-reverse'];
function CurrentBreakpoint() {
    var innerWidth = useWindowSize_1.useWindowSize().innerWidth;
    var breakpoint = useBreakpoint_1.useBreakpoint();
    return (react_2.default.createElement(react_2.default.Fragment, null,
        "width: ",
        innerWidth,
        "px | breakpoint: ",
        react_2.default.createElement("strong", null, breakpoint)));
}
function Cell(props) {
    return (react_2.default.createElement(__1.Cell, __assign({}, props),
        react_2.default.createElement("div", { style: styles.box }, props.children)));
}
react_1.storiesOf('Components|Grid', module)
    .add('default', function () { return (react_2.default.createElement(react_2.default.Fragment, null,
    react_2.default.createElement(CurrentBreakpoint, null),
    react_2.default.createElement(__1.Grid, { wrap: true },
        react_2.default.createElement(Cell, { xs: 12 }, "xs=12"),
        react_2.default.createElement(Cell, { xs: 12, sm: 6 }, "xs=12 sm=6"),
        react_2.default.createElement(Cell, { xs: 12, sm: 6 }, "xs=12 sm=6"),
        react_2.default.createElement(Cell, { xs: 12, sm: 6, md: 4 }, "xs=12 sm=6 md=4"),
        react_2.default.createElement(Cell, { xs: 12, sm: 6, md: 4 }, "xs=12 sm=6 md=4"),
        react_2.default.createElement(Cell, { xs: 12, sm: 6, md: 4 }, "xs=12 sm=6 md=4"),
        react_2.default.createElement(Cell, { xs: 12, sm: 6, md: 4, lg: 3 }, "xs=12 sm=6 md=4 lg=3"),
        react_2.default.createElement(Cell, { xs: 12, sm: 6, md: 4, lg: 3 }, "xs=12 sm=6 md=4 lg=3"),
        react_2.default.createElement(Cell, { xs: 12, sm: 6, md: 4, lg: 3 }, "xs=12 sm=6 md=4 lg=3"),
        react_2.default.createElement(Cell, { xs: 12, sm: 6, md: 4, lg: 3 }, "xs=12 sm=6 md=4 lg=3")))); })
    .add('flex-grow', function () { return (react_2.default.createElement(__1.Grid, null,
    react_2.default.createElement(Cell, { flexGrow: 1 }, "1"),
    react_2.default.createElement(Cell, null, "2"),
    react_2.default.createElement(Cell, { flexGrow: 2 }, "3"),
    react_2.default.createElement(Cell, null, "4"))); })
    .add('playground', function () { return (react_2.default.createElement(react_2.default.Fragment, null,
    react_2.default.createElement(__1.Grid, { alignItems: addon_knobs_1.select('alignItems', alignItemsOptions, 'center'), justifyContent: addon_knobs_1.select('justifyContent', justifyContentOptions, 'center'), direction: addon_knobs_1.select('direction', directionOptions, 'row'), gap: addon_knobs_1.number('gap', 2), gapVertical: addon_knobs_1.number('gapCrossAxis', 1), wrap: true },
        react_2.default.createElement(Cell, { xs: 6 }, "xs=6"),
        react_2.default.createElement(Cell, { xs: 6 }, "xs=6"),
        react_2.default.createElement(Cell, { xs: 6 }, "xs=6")),
    react_2.default.createElement("hr", null),
    react_2.default.createElement("div", { style: { height: 300 } },
        react_2.default.createElement(__1.Grid, { alignItems: addon_knobs_1.select('alignItems', alignItemsOptions, 'center'), justifyContent: addon_knobs_1.select('justifyContent', justifyContentOptions, 'center'), direction: addon_knobs_1.select('direction', directionOptions, 'row'), style: { height: '100%' } },
            react_2.default.createElement(Cell, { xs: 1 }, "1"),
            react_2.default.createElement(Cell, { xs: 1 },
                react_2.default.createElement("div", { style: { height: 60 } }, "2")),
            react_2.default.createElement(Cell, { xs: 1 }, "3"))))); })
    .add('cell', function () { return (react_2.default.createElement(__1.Grid, { alignItems: 'flex-start' },
    react_2.default.createElement(Cell, { xs: 1 }, "1"),
    react_2.default.createElement(Cell, { xs: 1 },
        react_2.default.createElement("div", { style: { height: 60 } }, "2")),
    react_2.default.createElement(Cell, { xs: addon_knobs_1.number('xs', 2), alignSelf: addon_knobs_1.select('alignSelf', alignSelfOptions, 'auto') },
        react_2.default.createElement("strong", null, "3")),
    react_2.default.createElement(Cell, { xs: 1 }, "4"))); });
